plugins/woocommerce-blocks/assets/js/atomic/utils/register-product-block-type.ts

```diff
@@ -60,7 +62,13 @@ export class BlockRegistrationManager {
 	/** Singleton instance of the manager */
 	private static instance: BlockRegistrationManager;
 	/** Map storing block configurations keyed by block name or variation name */
-	private blocks: Map< string, ProductBlockConfig > = new Map();
+	private blocks: Map<
+		string,
+		ProductBlockSettings & {
+			blockName: string;
+			settings: Record< string, unknown >;
+		}
```

Comment:
I didn't know how to keep this specific to `ProductBlockConfig` as it now depends on a generic that we don't know when building `BlockRegistrationManager`. A bit unfortunate, but given that all the methods used to interact with this property use more specific types, I think it shouldn't be a big deal. However, I'm open to suggestions on how to improve this. :slightly_smiling_face: